### PR TITLE
Feat/config aware poll

### PR DESCRIPTION
**Pull request by vird**
includes #14, #15

### CODE REVIEW

File: main/background.ts
Comment (hlolli) [inline main/background.ts]: I'm a fan of this eslint rule https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md in these sorts of situation, because it's less code to just write

```ts
let cachedMetricsTimeout: NodeJS.Timeout;
```

without assignment, it defaults to `NodeJS.Timeout | undefined`, null I feel should be used only when it's not a viable alternative to undefined, that's just opionated reply and you don't have to take this comment seriously

File: main/background.ts
Comment (hlolli) [inline main/background.ts:79]: you'll need find here, this is an array of objects with ids, you could reduce the array to an object of `Record<id, config>`

File: main/config.ts
Comment (hlolli) [inline main/config.ts]: I would go for more descriptive name

**Review by hlolli — APPROVED** [review]
early in development, nothing serious in my comments to not approve

File: main/background.ts
Comment (vird) [reply main/background.ts]: I need explicit null because I need clear this timeout on ping

Comment (hlolli) [reply main/background.ts]: but you can assign undefined to it?

File: main/background.ts
Comment (hlolli) [inline main/background.ts:65]: fantascit, just fyi these are implicitly 100% equivalent

```ts
let cachedMetricsTimeout: NodeJS.Timeout;
```

**Review by hlolli — APPROVED** [review]
I already approved, in case you were waiting for second approval I approve again, merge any time

**Pull request merged**
